src/components/Header/Header.js

```diff
@@ -0,0 +1,37 @@
+import { MDCTopAppBar } from "@material/top-app-bar";
+import headerTemplate from "./Header.html";
+
+export class HeaderComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.openFunction = props.openFunction;
+    this.scrollTarget = props.scrollTarget;
+  }
```

Comment:
1 - this.props = props --- and then use this.props.scrollTarget in code.
2 - we do not call callbacks Function - this should be not `openFunction ` but `onOpen` - it is an event